src/rgw/rgw_aws_lambda.cc

```diff
@@ -0,0 +1,116 @@
+//
+// Created by prateek on 30/03/20.
+//
+
+#include <string>
+#include <iostream>
+#include <aws/core/Aws.h>
+#include <aws/core/utils/Outcome.h>
+#include <aws/lambda/LambdaClient.h>
+#include <aws/lambda/model/InvokeRequest.h>
+
+namespace lambda {
+    const char *ALLOCATION_TAG = "TEST";
+    static std::shared_ptr<Aws::Lambda::LambdaClient> client;
+
+}
+
+#include <boost/lockfree/queue.hpp>
+#include <string>
+#include <utility>
+#include <aws/sns/SNSClient.h>
+#include <aws/sns/model/PublishRequest.h>
+
+namespace lambda {
+
+    static const int STATUS_OK = 1;
+    static const int STATUS_FULL = 2;
+    static const int STATUS_MANAGER_STOPPED = 3;
+
+    struct message_t {
+        std::string function_name;
+        std::string payload;
+
+        message_t(std::string function_name, std::string payload) : function_name(std::move(function_name)),
+                                                                    payload(std::move(payload)) {};
+    };
+
+    typedef boost::lockfree::queue<message_t *, boost::lockfree::fixed_sized<true>> MessageQueue;
+
+    class Manager {
+    private:
+        MessageQueue messageQueue;
+        std::thread runner;
+        bool stopped;
+
+        void publish_internal(message_t *msg) {
+          std::cout << "Invoking.." << std::endl;
+          Aws::Client::ClientConfiguration configuration;
```

Comment:
configuration can probably be a data member of `Manager`
what other parameters can be set inside configuration?